### PR TITLE
714-Create-builder-for-SpActionBarPresenter

### LINKED ISSUE #713: Add meta-data related to position in SpActionBarPresenter to position commands either left or right
Like that, we can create a builder for such presenter.

### LINKED ISSUE #714: Create builder for SpActionBarPresenter

### PR DESCRIPTION
**Pull request by juliendelplanque**
Added builder for SpActionBarPresenter and allowed to set position (left or right) in toolbar presenter as well.

Fixes #713
Fixes #714

### CODE REVIEW

File: src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	#name : #SpLeftPositionStrategyTest,
+	#superclass : #TestCase,
+	#category : #'Spec2-Commander2-Tests'
+}
+
+{ #category : #test }
+SpLeftPositionStrategyTest >> testAddButtonToActionBar [
+	| actionBar button |
+	actionBar := SpActionBarPresenter new.
+	button := ButtonPresenter new.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st]:
```suggestion
	button := SpButtonPresenter new.
```

File: src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	
+	self assert: actionBar items isEmpty.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st]:
```suggestion
	self assertEmpty: actionBar items.
```

File: src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	self assert: (actionBar items at: #start) anyOne equals: button
+]
+
+{ #category : #test }
+SpLeftPositionStrategyTest >> testAddButtonToToolbar [
+	| toolbar button |
+	toolbar := SpToolBarPresenter new.
+	button := SpToolBarButton new.
+	
+	self assert: toolbar items isEmpty.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st]:
```suggestion
	self assertEmpty: toolbar items.
```

File: src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	
+	SpLeftPositionStrategy new
+		addButton: button toToolbar: toolbar.
+		
+	self assert: toolbar leftItems size equals: 1.
+	self assert: toolbar leftItems anyOne equals: button.
+	self assert: toolbar rightItems size equals: 0.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpLeftPositionStrategyTest.class.st]:
```suggestion
	self assertEmpty: toolbar rightItems
```

File: src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	#name : #SpRightPositionStrategyTest,
+	#superclass : #TestCase,
+	#category : #'Spec2-Commander2-Tests'
+}
+
+{ #category : #test }
+SpRightPositionStrategyTest >> testAddButtonToActionBar [
+	| actionBar button |
+	actionBar := SpActionBarPresenter new.
+	button := ButtonPresenter new.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st]:
```suggestion
	button := SpButtonPresenter new.
```

File: src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	
+	self assert: actionBar items isEmpty.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st]:
```suggestion
	self assertEmpty: actionBar items.
```

File: src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	self assert: (actionBar items at: #end) anyOne equals: button
+]
+
+{ #category : #test }
+SpRightPositionStrategyTest >> testAddButtonToToolbar [
+	| toolbar button |
+	toolbar := SpToolBarPresenter new.
+	button := SpToolBarButton new.
+	
+	self assert: toolbar items isEmpty.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st]:
```suggestion
	self assertEmpty: toolbar items.
```

File: src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+	
+	SpRightPositionStrategy new
+		addButton: button toToolbar: toolbar.
+		
+	self assert: toolbar rightItems size equals: 1.
+	self assert: toolbar rightItems anyOne equals: button.
+	self assert: toolbar leftItems size equals: 0.
```
Comment (jecisc) [inline src/Spec2-Commander2-Tests/SpRightPositionStrategyTest.class.st]:
```suggestion
	self assertEmpty: toolbar leftItems
```

File: src/Spec2-Commander2/SpCommand.class.st [hunk 9]
```diff
@@ -121,7 +131,13 @@ SpCommand >> iconProvider: anObject [
 SpCommand >> initialize [ 
 
 	super initialize.
-	self beDisabledWhenCantBeRun
+	self beDisabledWhenCantBeRun.
```
Comment (demarey) [inline src/Spec2-Commander2/SpCommand.class.st]: should be `self beDisableWhenCannotBeExecuted`

Comment (juliendelplanque) [reply src/Spec2-Commander2/SpCommand.class.st]: Yes, but for another issue

**Pull request merged**